src/idlcxx/src/streamers.c

```diff
@@ -664,6 +672,73 @@ process_case_label(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+process_typedef(
+  const idl_pstate_t* pstate,
+  const bool revisit,
+  const idl_path_t* path,
+  const void* node,
+  void* user_data)
+{
+  (void)revisit;
+  (void)path;
+
+  const char* fmt =
+    "template<typename T>\n"
+    "void %2$s(T& str, %1$s& instance)\n{\n";
+  const char* constfmt =
+    "template<typename T>\n"
+    "void %2$s(T& str, const %1$s& instance)\n{\n";
+  const char* closefmt = "}\n\n";
+  char* name = NULL;
+
+  struct streams* streams = user_data;
+  idl_typedef_t* td = (idl_typedef_t*)node;
+  const idl_declarator_t* declarator;
+  idl_retcode_t ret = IDL_RETCODE_OK;
+
+  IDL_FOREACH(declarator, td->declarators) {
+    if (IDL_PRINTA(&name, get_cpp11_fully_scoped_name, declarator, streams->generator) < 0)
+    {
+      ret = IDL_RETCODE_NO_MEMORY;
+      break;
```

Comment:
Just return directly here, much easier. Same goes for the rest of the code. Thanks.

Reply:
will do